[Shuang Yuan] iP

shuang26 opened the pull request:
### Cow 🐄 
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Cow frees your mind of having to remember things you need to do. It's,

* **easy** to learn and use
* text-based

All you need to do is,

1. add your tasks (todo, deadline, event task types)
2. let it manage your task 

**Features:**
- [X] Adding tasks
- [ ] Deleting tasks
- [ ] Mark/Unmark tasks as done


Here is the _main_ method:
```ruby
public static void main(String[] args) {
    new Cow("data/cow.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

shuang26 committed:
Level-0

shuang26 committed:
Level-1

shuang26 committed:
Level-2

shuang26 committed:
Level-3

shuang26 committed:
Level-4

shuang26 committed:
Level-4 with changes to main class

shuang26 committed:
A-TextUiTesting, added tests for user commands

shuang26 committed:
Level-5, added error messages for most user command

shuang26 committed:
Level-6, added delete

shuang26 committed:
Created new class called TaskManager, and update tasks and its subclass

shuang26 committed:
Added saveToFile function to save when user exists program.

shuang26 committed:
Merge branch 'branch-Level-7'

shuang26 committed:
Add date and time into Deadline task

shuang26 committed:
Merge branch 'branch-Level-8'

shuang26 committed:
Add Ui class to refactor code in main class

shuang26 committed:
Update Cow and Ui class

shuang26 committed:
Add Storage and TaskList classes to seperate responsibilities

shuang26 committed:
Update main class

shuang26 committed:
Merge branch 'branch-A-MoreOOP'

shuang26 committed:
Add parser, storage, task, ui classes into their respective packages

shuang26 committed:
Add packages

shuang26 committed:
Remove TaskManager class and move its logic into Parser class

shuang26 committed:
Change data file path for storage

shuang26 committed:
Merge branch 'add-gradle-support'

shuang26 committed:
Add JUnit test for TaskList class

shuang26 committed:
Add JUnit test for storage

shuang26 committed:
Merge branch 'branch-A-JUnit'

shuang26 committed:
Add JavaDoc to all files in Task folder

shuang26 committed:
Add JavaDoc for Parser, Storage, Ui and Main class

shuang26 committed:
Add Find feature to TaskList class

LF1928 commented on src/main/java/Cow.java:
```diff
@@ -0,0 +1,54 @@
+            if (input.isEmpty()) {
+                continue;
+            }
+            if (tasks.handleInput(input)) {
```
I like how you had handleInput in TaskList

LF1928 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,66 @@
+     * @throws DateTimeParseException if the input string does not match the expected format.
+     */
+    public static LocalDateTime parseDateTime(String input) throws DateTimeParseException {
+        input = input.replaceAll("[-/]", " "); // Normalize separators
```
I like how you accounted for different separators

LF1928 commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,45 @@
+ * An Event task contains a description, completion status, and start and end dates.
+ */
+public class Event extends Task {
+    protected String fromDate;
+    protected String toDate;
```
maybe dates can be of type DateTime?

LF1928 commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,81 @@
+     *
+     * @return {@code true} if the task is done, {@code false} otherwise.
+     */
+    public boolean isDone() {
```
maybe change to getIsDone?

LF1928 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,324 @@
+     * @param isDone The completion status of the task.
+     * @return The created Todo task.
+     */
+    public Todo createTodo(String description, boolean isDone) {
```
Its nice that you split the create tasks

LF1928 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,324 @@
+    /**
+     * Prints out all the tasks in the task list.
+     */
+    public void listAllTask() {
```
change to listAllTasks?

shuang26 committed:
Add Command class and its subclasses and Refactor method and variable names after PR.

shuang26 committed:
Change Event task to support date and time format instead of String

shuang26 committed:
Fix command result when printing to screen

shuang26 committed:
Add JavaDoc for command classes

shuang26 committed:
Add CheckStyle and Edit classes with checkstyle error

shuang26 committed:
Fix checkstyle errors relating to testing files

shuang26 committed:
Add dependency to support JavaFX

Add resource folder with cs, images and view

shuang26 committed:
Add Launcher, Main, MainWindow and DialogBox class

JavaFx now works with the user input from GUI.
Remove the use of Ui class in Cow

shuang26 committed:
Modify logic to exit program when user input bye

Add getCommandType method to Command and its subclasses.
Add greeting message to user when they first open
the app.

shuang26 committed:
Change greeting message

shuang26 committed:
Merge branch 'branch-Level-10'

Add JavaFx and GUI and move the handling of user input
from CLI to GUI interface.

shuang26 committed:
Add JUnit tests for TaskList, Deadline and Event

Added test cases to ensure correct functionality of core operations
after modifying TaskList, Deadline, and Event classes in the previous commit.

These tests improve code reliability and maintainability

shuang26 committed:
Fix code to comply with Checkstyle rules

Reformatted code to meet indentation, and style guidelines

shuang26 committed:
Change toString method in deadline task

shuang26 committed:
Improve code quality through refactoring

Refactored code to enhance readability and maintainability.

Changes include:
- Applied consistent formatting and indentation.
- Refactored methods for better modularity and reusability.

These improvements enhance code clarity, reduce potential errors, and
make future modifications easier.

shuang26 committed:
Use Java Streams for searching of task in TaskList

shuang26 committed:
Merge pull request #2 from shuang26/A-Streams

A streams

shuang26 committed:
Refactor and Improve readability of TaskList class

Applied SLAP to improve code readability and maintainability.
Extracted methods to encapsulate logic and reduce complexity.
Improved error handling and validation checks.
Reformatted code for consistency and clarity.

shuang26 committed:
Improve code quality for Storage

Applied SLAP to improve code readability and maintainability.
Added test for ParserClass.

shuang26 committed:
Improve UI with new background and text colours

Adjusted text field and scroll bar styles for improved readability.
Adjusted button colors for improved look.

shuang26 committed:
Merge branch 'branch-A-BetterGui'

shuang26 committed:
Add Ui.png and Update README with Cow feature

shuang26 committed:
Add window title for Main class
